bindings/pyroot_experimental/cppyy/CPyCppyy/CMakeLists.txt

```diff
@@ -0,0 +1,48 @@
+set(headers
+  include/TPyArg.h
+  include/TPyException.h
+  include/TPyReturn.h
+  include/TPython.h
+)
+
+set(sources
+  src/CallContext.cxx
+  src/Converters.cxx
+  src/CPPClassMethod.cxx
+  src/CPPConstructor.cxx
+  src/CPPDataMember.cxx
+  src/CPPFunction.cxx
+  src/CPPInstance.cxx
+  src/CPPMethod.cxx
+  src/CPPOverload.cxx
+  src/CPPScope.cxx
+  src/CPPSetItem.cxx
+  src/CPyCppyyModule.cxx
+  src/CustomPyTypes.cxx
+  src/Executors.cxx
+  src/LowLevelViews.cxx
+  src/MemoryRegulator.cxx
+  src/ProxyWrappers.cxx
+  src/PyStrings.cxx
+  src/Pythonize.cxx
+  src/TemplateProxy.cxx
+  src/TPyArg.cxx
+  src/TPyClassGenerator.cxx
+  src/TPyException.cxx
+  src/TPyReturn.cxx
+  src/TPython.cxx
+  src/TupleOfInstances.cxx
+  src/TypeManip.cxx
+  src/Utility.cxx
+)
+
+add_library(cppyy SHARED ${headers} ${sources})
```

Comment:
This might clash with other `libcppyy.so` builds. Can we rename this to `ROOTcppyy`?

Reply:
Changing the name of `libcppyy.so` would make Cppyy fail. If you check (this line)[https://github.com/amadio/root/blob/cppyy/bindings/pyroot_experimental/cppyy/cppyy/python/cppyy/_cpython_cppyy.py#L19] in Cppyy's Python code, it is trying to import the extension module and use it from Python. For that line to work, it needs that a library called `libcppyy.so` is in the `PYTHONPATH`.

A similar thing happens with the other Cppyy library, `libcppyy_backend.so`, and (this line)[https://github.com/amadio/root/blob/cppyy/bindings/pyroot_experimental/cppyy/cppyy-backend/clingwrapper/python/cppyy_backend/loader.py#L9].

Reply:
I am predicting a mess of runtime / environment fragility. We can address this when we get there.

(We *know* we want load load *our* library and that we do not want to interfere with the users' pip stuff etc; we should dial our library very explicitly without having to ask Python to resolve anything. This even to the extreme that we should not even *have* separate dynamic libraries for this: they should all be part of one single `libROOTPyROOT` or whatever you call it.)